kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ValidationResult.java

```diff
@@ -0,0 +1,73 @@
+package org.eclipse.kura.container.signature;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * Class representing the result of the signature validation performed by {@link:ContainerSignatureValidationService}
+ * 
+ * The result of the validation is composed of two main parts: whether or not the container image signature was
+ * validated and the container image digest (in the "algorithm:encoded" format, @see
+ * <a href="https://github.com/opencontainers/image-spec/blob/main/descriptor.md#digests">Opencontainers specs</a>)
+ *
+ * If the signature is valid, the image digest MUST be provided.
+ *
+ * @since 2.7
+ */
+@ProviderType
+public final class ValidationResult {
+
+    private boolean isSignatureValid = false;
+    private Optional<String> imageDigest = Optional.empty();
+
+    public ValidationResult() {
+        // Nothing to do
+    }
+
+    public ValidationResult(boolean signatureValid, String digest) {
+        this.imageDigest = Optional.of(Objects.requireNonNull(digest));
+        this.isSignatureValid = Objects.requireNonNull(signatureValid);
+
+        if (this.isSignatureValid && (!this.imageDigest.isPresent() || this.imageDigest.get().isEmpty())) {
```

Comment:
Does it make sense to have it in line 30 before the assignments?

Reply:
Addressed in 8ff5490112d50df1415e7dfd1bb526a0baa247fb